test/mini_racer_test.rb

```diff
@@ -283,6 +283,13 @@ def test_return_unknown
     assert_equal("Undefined Conversion", context.eval("test()"))
   end
 
+  def test_fatal_alloc
+    context = MiniRacer::Context.new(max_memory: 1000000000)
```

Comment:
I would soft limit at 200mb cause 1gb is a bit high for the test it is probably going to fail on travis at some point